test/test_producer.py

```diff
@@ -34,10 +34,15 @@ def test_end_to_end(kafka_broker, compression):
         elif platform.python_implementation() == 'PyPy':
             return
 
+    if compression == 'zstd' and env_kafka_version() < (2, 1, 0):
+        return
+    env_version = env_kafka_version()
+    api_version = env_version if env_version >= (2, 1, 0) else None
```

Comment:
What is the purpose of adding this? I'm not saying it's incorrect, I simply don't understand why the test didn't previously use this and then now needs it...

Reply:
the point was to force an api version injection only if the env var defined api si larger than 2.1, else let the connector handle it by querying the broker. Now that you mention it, i don't see how it could actually help, but i do remember some weird behaviour on a combination of detected broker version and actual broker version, since they don't always seem to match. I'll leave that part out and come back to it if i have a clear scenario when X does not play nicely with Y